core/store/src/main/java/mil/nga/giat/geowave/core/store/query/QueryOptions.java

```diff
@@ -379,7 +407,27 @@ public static long getSerialversionuid() {
 			}
 		}
 
-		final ByteBuffer buf = ByteBuffer.allocate(16 + authBytes.length + aSize + iSize);
+		int fieldIdsBytesNeeded = 0;
+		byte[] fieldIdsBytes = null;
+		if (fieldIds != null && fieldIds.size() > 0) {
+			final StringBuffer sBuf = new StringBuffer();
+			final String comma = ",";
+			for (String fieldId : fieldIds) {
+				if (sBuf.length() > 0) {
+					sBuf.append(comma);
+				}
+				sBuf.append(fieldId);
+			}
+			fieldIdsBytes = StringUtils.stringToBinary(sBuf.toString());
+			fieldIdsBytesNeeded = 4 + fieldIdsBytes.length;
+		}
+
+		final ByteBuffer buf = ByteBuffer.allocate(20 + authBytes.length + aSize + iSize + fieldIdsBytesNeeded);
+		buf.putInt((fieldIdsBytesNeeded > 0) ? 1 : 0);
```

Comment:
Do we need the extra int, or can we just place a 0 if the field id list is emty?